hail/src/main/scala/is/hail/utils/package.scala

```diff
@@ -624,6 +624,35 @@ package object utils extends Logging
       s
   }
 
+  trait ReadOnlyMap[K, V] extends (K => V) {
+    def apply(k: K): V
+    def contains(k: K): Boolean
+    def get(k: K): Option[V]
+    def getOrElse(k: K, v: V): V
+  }
+
+  class ReadOnlyMutableMap[K, V](m: mutable.Map[K, V])
```

Comment:
I think you can use `collection.Map` to the same effect -- build a mutable map, return it, and it satisfies that type without exposing the mutability.

Reply:
oh, nice, I was really annoyed that the default `Map` interface was incompatible with `mutable.Map`

Reply:
this is one of the weird dark corners of the scala standard collections for sure